src/main/java/org/truffleruby/core/rope/Rope.java

```diff
@@ -108,6 +106,14 @@ public final boolean isHashCodeCalculated() {
         return hashCode != 0;
     }
 
+    public final int calculatedHashCode() {
+        return hashCode;
+    }
+
+    public final boolean hashesMatch(Rope other) {
+        return (isHashCodeCalculated() && hashCode == other.hashCode) || (hashCode() == other.hashCode());
```

Comment:
This will compute the hashCode if it's not already computed, which can be expensive for equal calls as it makes a second iteration over all bytes.